File: budget-app-fe/src/app/classes/dto/filtering/query-group.ts
import {FilterTypeEnum} from './filter-type.enum';
import {IComparableItem} from './interfaces/comparable-item.interface';

export class QueryGroup implements IComparableItem {
  typeDiscriminator: string = 'QueryGroup';
  public comparisonType: FilterTypeEnum = FilterTypeEnum.none;
  public queries: Array<IComparableItem> = [];
}

File: BudgetApp.Backend/BudgetApp.Backend.Dto/Filtering/QueryGroup.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;
using BudgetApp.Backend.Dto.Interfaces;

namespace BudgetApp.Backend.Dto.Filtering
{
    public class QueryGroup : IComparableItem
    {
        [JsonPropertyName("comparisonType")]
        public FilterType? ComparisonType { get; set; }

        [JsonPropertyName("typeDiscriminator")]
        public string TypeDiscriminator => nameof(QueryGroup);

        [JsonPropertyName("queries")]
        public List<IComparableItem> Queries { get; set; } = new();
    }
}